pyogrio/tests/test_path.py

```diff
@@ -85,6 +86,8 @@ def change_cwd(path):
             "s3://testing/test.zip!a/b/item.shp",
             "/vsizip/vsis3/testing/test.zip/a/b/item.shp",
         ),
+        ("/vsimem/data.gpkg", "/vsimem/data.gpkg"),
+        (Path("/vsimem/data.gpkg"), "/vsimem/data.gpkg"),
```

Comment:
I am not entirely sure we should allow `Path` objects with VSI file paths (that's not what they are meant for), but I suppose this is just supported because we all Paths in general as input, and not because we support this use case explicitly?

Reply:
Although seeing the test `test_get_vsi_path_or_buffer_obj_to_string` just below, indicating that we explicitly ensure forward slashes are preserved, this support is actually quite explicit?

Reply:
I don't quite know what you mean. But, anyway, we cannot control how a user passes in the path (as a Path or as a str)...

Reply:
We cannot control what the user passes, but that doesn't mean we have to accept everything the user passes, and so we _could_ not accept `Path("/vsimem/data.gpkg")`, because that is mixing two concepts (Python file path objects and GDAL VSI paths)

Reply:
Ah, ok, now I understand: throw an error if a Path("/vsimem/") is passed in. Yes, that is an option as well I suppose, even though I personally viewed "/vsimem/..." conceptionally as a "path" myself, hence my confusion.

As it is about the same effort to support it I don't really see the advantage in doing that?

Reply:
They are indeed also file paths, but not _OS_ file _system_ paths. You can indeed still use them for the nice concatenation functionality etc (like `Path(..) / ".."`), but even on Windows that essentially already breaks down when converting that naively into the string representation. So I would say it is a bit misuse of the pathlib library.

(in theory, pathlib also provides a `PurePath` that just does the path manipulation operations without any actual file system or IO operations, so that would be better fitting here. But only allowing that is probably only more confusing for users)

Now, if it is almost as much work to disallow as to allow, I won't argue strongly about disallowing them.